presto-main/src/main/java/com/facebook/presto/connector/system/SystemFunctions.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.connector.system;
+
+import com.facebook.presto.client.NodeVersion;
+import com.facebook.presto.metadata.InternalNodeManager;
+import com.facebook.presto.metadata.PrestoNode;
+import com.facebook.presto.metadata.Signature;
+import com.facebook.presto.metadata.SqlScalarFunction;
+import com.facebook.presto.spi.Node;
+import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableList;
+import io.airlift.slice.Slice;
+import io.airlift.slice.Slices;
+
+import javax.inject.Inject;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.reflect.Method;
+
+import static com.facebook.presto.metadata.FunctionKind.SCALAR;
+import static com.facebook.presto.spi.type.VarcharType.VARCHAR;
+import static java.lang.invoke.MethodHandles.lookup;
+import static java.util.Objects.requireNonNull;
+
+public final class SystemFunctions
+{
+    private final InternalNodeManager nodeManager;
+
+    @Inject
+    public SystemFunctions(InternalNodeManager nodeManager)
+    {
+        this.nodeManager = requireNonNull(nodeManager, "nodeManager is null");
+    }
+
+    private Slice getVersion()
+    {
+        Node node = nodeManager.getCurrentNode();
```

Comment:
This is brittle. If a cluster has nodes with mixed versions, the result of the query will vary depending on where it's scheduled. A more robust implementation would be to record the version in the query session in the same way "current timestamp" is stored, so that all worker nodes get a consistent view.